File: bj_17389.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int num = Integer.parseInt(br.readLine());
		String str = br.readLine();
		int res = 0;
		int bns = 0;
		for(int i=0; i<num; i++) {
			char score = str.charAt(i);
			if(score == 'O') { res += i+1+bns; bns++;}
			else if(score == 'X') bns = 0;
		}
		System.out.println(res);
	}
}

File: bj_17389.kt
package com

import java.io.*

fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val num = br.readLine().toInt()
    val str = br.readLine()
    var bns = 0
    var res = 0
    for(i in 0..<num){
        val cr = str.get(i)
        if(cr == 'O'){
            res += i+1+bns
            bns++
        }
        else if(cr == 'X'){bns=0}
    }
    println(res)
}